test/ads.jl

```diff
@@ -0,0 +1,129 @@
+using TaylorIntegration
+using Test
+
+@testset "Automatic Domain Splitting" begin
+    using TaylorIntegration: ADSBinaryNode, countnodes, timesvector
+    using StaticArrays: SVector
+
+    # This example is based upon section 3 of
+    # https://doi.org/10.1007/s10569-015-9618-3.
+
+    # Dynamical function
+    @taylorize function kepler_eqs!(dq, q, params, t)
+        dq[1] = q[3]
+        dq[2] = q[4]
+        rr = ( q[1]^2 + q[2]^2 )^(3/2)
+        dq[3] = - q[1] / rr
+        dq[4] = - q[2] / rr
+    end;
+    # Initial conditons (plain)
+    local q00 = [1.0, 0.0, 0.0, sqrt(1.5)]
+    # Jet transport variables
+    local dq = set_variables("dx dy", numvars = 2, order = 5)
+    # Initial conditions (jet transport)
+    local q0_ = q00 .+ [0.008, 0.08, 0.0, 0.0] .* vcat(dq, dq)
+    local q0 = SVector{4}(q0_)
+    # Jet transport domain
+    local dom = ADSDomain((-1.0, 1.0), (-1.0, 1.0))
+    # Initial time
+    local t0 = 0.0
+    # Final time
+    local tmax = 34.85
+    # Taylor1 order
+    local order = 25
+    # Splitting tolerance
+    local stol = 0.008
+    # Absolute tolerance
+    local abstol = 1e-20
+    # Dynamical function parameters
+    local params = nothing
+    # Maximum allowed splits
+    local maxsplits = 25
+    # Maximum allowed steps
+    local maxsteps = 1_000
```

Comment:
Since `@testset begin ... end` introduces a scope, I think it's safe to remove the `local`s here.

Reply:
Done.